src/util.test.ts

```diff
@@ -1,56 +1,23 @@
-import { basicAuth, noCache, parseGitName } from './util';
+import { parseBasicAuth, BasicAuthError, noCache, parseGitName } from './util';
 
 describe('util', () => {
-  describe('basicAuth', () => {
-    test('should send back basic auth headers', (done) => {
-      const headers: any = {};
-
+  describe('parseBasicAuth', () => {
+    test('should throw error if headers invalid or missing', () => {
       const req: any = {
         headers: {},
       };
 
-      const res: any = {
-        writeHead: function (_code: number) {
-          code = _code;
-        },
-        setHeader: function (key: string | number, value: any) {
-          headers[key] = value;
-        },
-        end: function (_status: number) {
-          status = _status;
-          expect(code).toBe(401);
-          expect(headers).toEqual({
-            'Content-Type': 'text/plain',
-            'WWW-Authenticate': 'Basic realm="authorization needed"',
-          });
-          expect(status).toBe('401 Unauthorized');
-          done();
-        },
-      };
-
-      let code = 0;
-      let status = 0;
-
-      basicAuth(req, res, () => {
-        expect('').not.toEqual('should not have entered this callback');
-        done();
-      });
+      expect(parseBasicAuth.bind(null, req)).toThrow(new BasicAuthError());
```

Comment:
can we also keep the status code asserts? We should make sure the response is formed correctly in the event of an error.

Reply:
yeah absolutely those are the kind of tests that have made me stay in WIP